x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/field_select.tsx

```diff
@@ -73,21 +71,21 @@ export function FieldSelect({
 
     const [specialFields, normalFields] = _.partition(
       fields,
-      (field) => fieldMap[field].type === 'document'
+      (field) => currentIndexPattern.getFieldByName(field)?.type === 'document'
     );
 
     const containsData = (field: string) =>
-      fieldMap[field].type === 'document' ||
+      currentIndexPattern.getFieldByName(field)?.type === 'document' ||
       fieldExists(existingFields, currentIndexPattern.title, field);
 
     function fieldNamesToOptions(items: string[]) {
       return items
         .map((field) => ({
-          label: fieldMap[field].displayName,
```

Comment:
This doesn't seem like a good handling of the error case here (it could create entries without a label) - let's filter out all fields without entry in the field map instead before mapping

Reply:
How would that happen? Fields are validated against the `containsData` here

Reply:
They are not for all invocations - `fieldNamesToOptions` is also called for empty fields in line 138 of this file.